scopes/compositions/compositions/compositions.tsx

```diff
@@ -32,9 +32,13 @@ export type MenuBarWidget = {
   location: 'start' | 'end';
   content: ReactNode;
 };
-export type CompositionsProp = { menuBarWidgets?: CompositionsMenuSlot; emptyState?: EmptyStateSlot };
+export type CompositionsProp = {
+  menuBarWidgets?: CompositionsMenuSlot;
+  emptyState?: EmptyStateSlot,
+  customUI?: {prepend: boolean, component: (component: ComponentModel) => JSX.Element};
```

Comment:
no.
1. CustomUI === children
2. don't pass component as prop, they can use `ComponentContext` directly
3. prepend true/false is pain, just pass both options

how about this?
```
slots?: {
  menuItems?: ReactNode,
  preComposition?: ReactNode,
  postComposition?: ReactNode,
  end?: ReactNode // why do we need a slot after the split pane?
}
```

Reply:
1. Not sure what you mean
2. good point
3. sounds good

Re the slots comment, you want to just merge all the slots together, so remove the separate parameter for menuItems?
If so sounds fine to me (though not sure what the question is in the comments about the end slot - think you added that...)

Reply:
Though i can actually see a use for a content slot after the split pane. Something that's generic for all the components